src/Microsoft.Health.Fhir.Api/Controllers/OperationsController.cs

```diff
@@ -0,0 +1,128 @@
+using Microsoft.Health.Fhir.Api.Features.Filters;
+using Microsoft.Health.Fhir.Api.Features.Routing;
+using Microsoft.Health.Fhir.Core.Configs;
+using Microsoft.Health.Fhir.Core.Features.Context;
+
+namespace Microsoft.Health.Fhir.Api.Controllers
+{
+    [ServiceFilter(typeof(OperationOutcomeExceptionFilterAttribute))]
+    public class OperationsController : Controller
+    {
+        private readonly ILogger<OperationsController> _logger;
+        private readonly IFhirRequestContextAccessor _fhirRequestContextAccessor;
+        private readonly OperationsConfiguration _operationsConfig;
+
+        public OperationsController(
+            ILogger<OperationsController> logger,
+            IFhirRequestContextAccessor fhirRequestContextAccessor,
+            IOptions<OperationsConfiguration> operationConfig)
+        {
+            EnsureArg.IsNotNull(logger, nameof(logger));
+            EnsureArg.IsNotNull(fhirRequestContextAccessor, nameof(fhirRequestContextAccessor));
+            EnsureArg.IsNotNull(operationConfig, nameof(operationConfig));
+
+            _logger = logger;
+            _fhirRequestContextAccessor = fhirRequestContextAccessor;
+            _operationsConfig = operationConfig.Value;
+        }
+
+        [HttpGet]
+        [Route(KnownRoutes.Export)]
+        [ValidateOperationHeadersFilter]
+        public IActionResult Export()
+        {
+            HttpStatusCode returnCode;
+            OperationOutcome result;
+
+            if (_operationsConfig.SupportsExport)
+            {
+                result = GenerateOperationOutcome(
+                    OperationOutcome.IssueSeverity.Error,
+                    OperationOutcome.IssueType.NotSupported,
+                    "Export operation not supported");
+                returnCode = HttpStatusCode.NotImplemented;
+            }
+            else
+            {
+                result = GenerateOperationOutcome(
+                    OperationOutcome.IssueSeverity.Error,
+                    OperationOutcome.IssueType.Value,
+                    "Export operation disabled");
+                returnCode = HttpStatusCode.BadRequest;
+            }
+
+            return FhirResult.Create(result, returnCode);
+        }
+
+        [HttpGet]
+        [Route(KnownRoutes.ExportResourceType)]
+        [ValidateOperationHeadersFilter]
+        public IActionResult ExportResourceType(string type)
+        {
+            // Export by ResourceType is supported only for Patient resource type.
+            if (!string.Equals(type, "Patient", StringComparison.Ordinal))
+            {
+                OperationOutcome result = GenerateOperationOutcome(
+                    OperationOutcome.IssueSeverity.Error,
+                    OperationOutcome.IssueType.NotSupported,
+                    $"{type} type not supported for Export by ResourceType operation");
+
+                return FhirResult.Create(result, HttpStatusCode.BadRequest);
+            }
+
+            return Export();
```

Comment:
Are we falling back to this just to handle the logic on if it's enabled or not?

Reply:
Yes. This will change once we add the actual functionality. I will add some comments to the code regarding this.